spec/archive_spec.rb

```diff
@@ -0,0 +1,24 @@
+require "spec_helper"
+
+describe Alephant::Broker::LoadStrategy::S3::Archived do
+  subject { described_class.new }
+
+  describe "#s3_path" do
+    let(:id) { 42 }
+    let(:component_meta) { double(:id => id) }
+
+    specify do
+      expect(subject.s3_path component_meta).to eq id
+    end
+
+    context "no location associated with component meta" do
+      let(:component_meta) { double(:id => nil) }
+
+      specify do
+        expect do
+          subject.s3_path component_meta
+        end.to raise_error Alephant::Broker::InvalidCacheKey
+      end
```

Comment:
I think it's a little harder to read with the nested do/end, would it be nicer as:

```ruby
...
let(:expected_exception) { Alephant::Broker::InvalidCacheKey }

specify do
  expect { subject.s3_path component_meta }. to raise_error expected_exception
end
```

?